src/Surfnet/StepupMiddlewareClient/Identity/Dto/RaSecondFactorSearchQuery.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddlewareClient\Identity\Dto;
+
+use Assert;
+
+final class RaSecondFactorSearchQuery implements HttpQuery
+{
+    const STATUS_UNVERIFIED = 'unverified';
+    const STATUS_VERIFIED = 'verified';
+    const STATUS_VETTED = 'vetted';
+    const STATUS_REVOKED = 'revoked';
+
+    /**
+     * @var string
+     */
+    private $institution;
+
+    /**
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @var string|null
+     */
+    private $type;
+
+    /**
+     * @var string|null The second factor type's ID (eg. Yubikey public ID)
+     */
+    private $secondFactorId;
+
+    /**
+     * @var string|null
+     */
+    private $email;
+
+    /**
+     * @var string|null One of the STATUS_* constants.
+     */
+    private $status;
+
+    /**
+     * @var string|null
+     */
+    private $orderBy;
+
+    /**
+     * @var string|null
+     */
+    private $orderDirection;
+
+    /**
+     * @var int
+     */
+    private $pageNumber;
+
+    /**
+     * @param string $institution
+     * @param int $pageNumber
+     */
+    public function __construct($institution, $pageNumber)
+    {
+        $this->assertNonEmptyString($institution, 'institution');
+        Assert\that($pageNumber)
```

Comment:
We've not been using Assert for this up to now, for consistency sake I'd rather not start using them now. A chore to refactor all `if(!is_*){ throw exception }` to asserts would be very welcome. Till then I'd opt for consistency.